clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -126,15 +134,15 @@ public ProduceRequest(ProduceRequestData produceRequestData, short version) {
     }
 
     // visible for testing
-    Map<TopicPartition, Integer> partitionSizes() {
+    Map<TopicIdPartition, Integer> partitionSizes() {
         if (partitionSizes == null) {
             // this method may be called by different thread (see the comment on data)
             synchronized (this) {
                 if (partitionSizes == null) {
-                    Map<TopicPartition, Integer> tmpPartitionSizes = new HashMap<>();
+                    Map<TopicIdPartition, Integer> tmpPartitionSizes = new HashMap<>();
                     data.topicData().forEach(topicData ->
                         topicData.partitionData().forEach(partitionData ->
-                            tmpPartitionSizes.compute(new TopicPartition(topicData.name(), partitionData.index()),
+                            tmpPartitionSizes.compute(new TopicIdPartition(topicData.topicId(), partitionData.index(), topicData.name()),
```

Comment:
will we ever need the name and ID for this data structure? I know fetch had something where we pass in a map to convert IDs to names if needed.

Just want to make sure folks won't use this info expecting the name to be there. If we don't think it is needed, maybe just include a comment about it.

Reply:
In the context of `partitionSizes` method we don't need topic name however, 
1. I didn't want to have if-else around this to decide if I will populate topic name as this logic will handle both old and new version
2. `TopicIdPartition` contractor need either a topic name to construct `TopicPartition` automatically or pass a constructed `TopicPartition` which need topic name as well.

Reply:
Ok -- it is fine if we don't need the name. But it might be good to leave a comment or something that says the name will not be present for older versions.